src/use-auth.js

```diff
@@ -0,0 +1,86 @@
+import React, {
+  useState,
+  useContext,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useQuery, useMutation, gql } from '@apollo/client';
+
+const REGISTER = gql`
+  mutation RegisterMutation(
+    $username: String!
+    $firstName: String!
+    $lastName: String!
+    $password: String!
+  ) {
+    createUser(
+      username: $username
+      firstName: $firstName
+      lastName: $lastName
+      password: $password
+      userRoleId: 3
+    ) {
+      id
+    }
+  }
+`;
+const QUERY_USERS = gql`
+{
+  users {
+    username
+    id
+    password
+  }
+}
+`;
+
+const authContext = createContext();
+
+const useAuth = () => useContext(authContext);
```

Comment:
`useAuth` can be a named export since it's a consuming hook, usually it's best to expose the main object or component as the default export and expose the utility like functions as named exports